VisualizationBase/src/Scene.cpp

```diff
@@ -175,24 +177,48 @@ bool Scene::switchToView(const QString viewName)
 	return view != nullptr;
 }
 
-void Scene::addViewItem(ViewItem *view)
+void Scene::addViewItem(ViewItem *view, QPoint position)
 {
 	Q_ASSERT(!inAnUpdate_);
-	viewItems_.append(view);
+	if (position.x() <= 0 || position.y() <= 0)
+		position = nextEmptyPosition();
+	if (viewItems_[position.x()].size() <= position.y())
+		viewItems_[position.x()].resize(position.y() + 1);
+	//If there already is an item -> use insert
+	if (viewItems_[position.x()][position.y()])
```

Comment:
I will add some boundary checks. However, I left any checks to enforce at maximum a 3x3 grid on purpose - I didn't see any reason to really enforce it as it doesn't break anything.